pkg/validate/validate.go

```diff
@@ -107,25 +107,63 @@ func ImagePullSecret(secret string) error {
 // the ClusterName function to confirm compliance with the rest. The resulting name will
 // therefore match RFC1035 with the exception of possible periods ".", which will be
 // translated into dashes "-" in the InfraID before being used to create cloud resources.
-func ClusterName1035(v string) error {
-	re := regexp.MustCompile("^[a-z]")
-	if !re.MatchString(v) {
-		return errors.New("cluster name must begin with a lower-case letter")
+// The optional param m is the max length for the string which the platform support.
+func ClusterName1035(v string, m ...int) error {
+
+	// checks the cluster name starts with a lower-case letter
+	err := ClusterStartName(v)
+	if err != nil {
+		return err
 	}
-	return ClusterName(v)
+
+	// checks if max length for the string was set
+	if m != nil {
+		err := ClusterName(v, m...)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := ClusterName(v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 // ClusterName checks if the given string is a valid name for a cluster and returns an error if not.
 // The max length of the DNS label is `DNS1123LabelMaxLength + 9` because the public DNS zones have records
 // `api.clustername`, `*.apps.clustername`, and *.apps is rendered as the nine-character \052.apps in DNS records.
-func ClusterName(v string) error {
-	maxlen := validation.DNS1123LabelMaxLength - 9
+// The optional param m is the max length for the string which the platform support.
+// Current default value for maxlen is validation.DNS1123LabelMaxLength - 9
+func ClusterName(v string, m ...int) error {
+	// Set the maximum length for the platform
+	maxlen := 0
+
+	// ClusterName has no m as param
+	if m == nil {
+		maxlen = validation.DNS1123LabelMaxLength - 9
+	} else if m[0] > 0 {
+		maxlen = m[0]
+	} else {
+		return errors.New("cluster name: unable to detect the max length")
+	}
+
 	if len(v) > maxlen {
 		return errors.New(validation.MaxLenError(maxlen))
 	}
 	return validateSubdomain(v)
 }
 
+// ClusterStartName checks the cluster name starts with a lower-case letter "^[a-z]"
+func ClusterStartName(v string) error {
```

Comment:
This function is only called internally to the package, is short, and is only called in one place. It does not warrant extracting into a function.